terraform-test/action.yml

```
@@ -0,0 +1,145 @@
+name: "Run terraform test"
+description: |
+  Terraform test for modules.
+author: "Artjoms Laivins"
+
+inputs:
+  test-file:
+    description: Test file to run.
+    required: true
+
+outputs:
+  exit-code:
+    description: The exit code from the terraform test command.
+    value: ${{ steps.run-tests.outputs.exit-code }}
+  summary:
+    description: Summary of the test run.
+    value: ${{ steps.run-tests.outputs.summary }}
+  report:
+    description: Report file of the test run.
+    value: ${{ steps.run-tests.outputs.report }}
+  json:
+    description: JSON output of the test run.
+    value: ${{ steps.run-tests.outputs.json }}
+
+
+runs:
+  using: "composite"
+  steps:
+    - id: check-prereqs
+      shell: bash
+      run: |
+        # Make sure terraform is available
+
+        set -o allexport; source "${{ github.action_path }}/helpers.sh"; set +o allexport;
+
+        start-group "check prerequisite: terraform binary"
+        if ! command -v terraform version &>/dev/null; then
+          log-error "terraform is not available on path, please install using the 'setup-terraform' action!"
+          exit 1
+        else
+          log-info "using $(terraform version)"
+        fi
+        end-group
+
+    - id: azure-login
+      uses: azure/login@v2
+      with:
+         tenant-id: ${{ env.ARM_TENANT_ID }}
+         subscription-id: ${{ env.ARM_SUBSCRIPTION_ID }}
+         client-id: ${{ env.ARM_CLIENT_ID }}   
+
+    - id: run-tests
+      working-directory: ${{ github.workspace }}
+      shell: bash
+      continue-on-error: true
+      env:
+        TF_IN_AUTOMATION: true
+      run: |
+        set -o allexport; source "${{ github.action_path }}/helpers.sh"; set +o allexport;
+
+        TEST_RESULT_JSON="${GITHUB_WORKSPACE}/${{ github.run_id }}-${{ inputs.test-file }}-$(date +%Y%m%d%H%M%S)-test-results.json"
+        TEST_REPORT_FILE="${GITHUB_WORKSPACE}/${{ github.run_id }}-${{ inputs.test-file }}-$(date +%Y%m%d%H%M%S)-test-report.txt"
+        TEST_CMD="terraform test --filter=tests/${{ inputs.test-file}} -json"
+
+        log-info "Running test command: ${TEST_CMD}"
+        start-group "'terraform test' "
+        #Run test
+        set -o pipefail
+          
+        set +e
+        ${TEST_CMD} | tee ${TEST_RESULT_JSON}
+        TEST_EXIT_CODE=${?}
+
+        log-multiline "Final JSON output" "$TEST_RESULT_JSON"
```

did you mean to cat the file maybe?

```suggestion
        log-multiline "Final JSON output" "$(cat $TEST_RESULT_JSON)"
```

---

yes, it was used during DEBUG sessions actually. Shall we behold this logging ?